### PR TITLE
dev-games/irrlicht: version bump to 1.8.4.

### PR DESCRIPTION
**Pull request by Hawk777**
Hello @gentoo/games,
There is a new version of Irrlicht out. Copying and pasting the ebuild appears to work. Repoman warnings are due to the file size of demoMake.patch, which was not modified from its previous form.

### CODE REVIEW

File: dev-games/irrlicht/irrlicht-1.8.4.ebuild [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=5
```
Comment (SoapGentoo) [inline dev-games/irrlicht/irrlicht-1.8.4.ebuild]: please modernize to `EAPI=6`

File: dev-games/irrlicht/irrlicht-1.8.4.ebuild [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	x11-proto/xproto
+	x11-proto/xf86vidmodeproto"
+
+S=${WORKDIR}/${P}/source/Irrlicht
+
+src_prepare() {
+	cd "${WORKDIR}"/${P} || die
+	edos2unix include/IrrCompileConfig.h
+
+	epatch \
```
Comment (SoapGentoo) [inline dev-games/irrlicht/irrlicht-1.8.4.ebuild]: put these in the global `PATCHES` array, then call `default` somewhere in `src_prepare`

**Comment by monsieurp** [comment]
ping?

**Comment by Hawk777** [comment]
Will work on this when I have time. Maybe tomorrow evening. Thanks for the comments, SoapGentoo.

**Comment by Hawk777** [comment]
Rebased and updated to EAPI 6.

File: dev-games/irrlicht/irrlicht-1.8.4.ebuild [hunk 3]
```diff
@@ -0,0 +1,74 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+inherit eutils multilib toolchain-funcs
```
Comment (SoapGentoo) [inline dev-games/irrlicht/irrlicht-1.8.4.ebuild]: you don't need `multilib` for `get_libdir` in `EAPI=6` anymore

File: dev-games/irrlicht/irrlicht-1.8.4.ebuild [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+HOMEPAGE="http://irrlicht.sourceforge.net/"
+SRC_URI="mirror://sourceforge/irrlicht/${P}.zip"
+
+LICENSE="ZLIB"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc ~x86"
+IUSE="debug doc static-libs"
+
+RDEPEND="virtual/jpeg:0
+	media-libs/libpng:0
```
Comment (SoapGentoo) [inline dev-games/irrlicht/irrlicht-1.8.4.ebuild]: libpng needs subslot operator `:0=`

File: dev-games/irrlicht/irrlicht-1.8.4.ebuild [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+	x11-proto/xproto
+	x11-proto/xf86vidmodeproto"
+
+S=${WORKDIR}/${P}/source/Irrlicht
+
+PATCHES=( "${FILESDIR}"/${P}-gentoo.patch
+		"${FILESDIR}"/${P}-config.patch
+		"${FILESDIR}"/${P}-demoMake.patch
+		"${FILESDIR}"/${P}-mesa-10.x.patch
+		"${FILESDIR}"/${P}-jpeg-9a.patch )
```
Comment (SoapGentoo) [inline dev-games/irrlicht/irrlicht-1.8.4.ebuild]: too much indentation, use one tab only

File: dev-games/irrlicht/irrlicht-1.8.4.ebuild [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+	sys-libs/zlib
+	virtual/opengl
+	x11-libs/libX11
+	x11-libs/libXxf86vm"
+DEPEND="${RDEPEND}
+	app-arch/unzip
+	x11-proto/xproto
+	x11-proto/xf86vidmodeproto"
+
+S=${WORKDIR}/${P}/source/Irrlicht
```
Comment (SoapGentoo) [inline dev-games/irrlicht/irrlicht-1.8.4.ebuild]: Try replacing `Irrlicht` by `${PN^}`

**Comment by Hawk777** [comment]
Fixed.

**Comment by SoapGentoo** [comment]
LGTM, @gentoo/proxy-maint

**Comment by monsieurp** [comment]
Nope. Someone from the Games team should review it, not us.

**Comment by monsieurp** [comment]
Oh well. Let's get on with it.

**Comment by monsieurp** [comment]
LGTM as well.

**Comment by Hawk777** [comment]
@monsieurp it looks like when you committed this you changed the line endings in the patch file, which led to bug #593328.

Specifically, my irrlicht-1.8.4-mesa-10.x.patch is a UNIX-formatted patch file (i.e. the patch file itself has LF line endings) against a Windows-formatted target file (i.e. the file being patched has CRLF line endings). By the time it made its way into the main Gentoo repo, it had somehow lost all the CRs and therefore no longer applies.

What should I do?

**Comment by monsieurp** [comment]
Hm. This is a problem with my pram script. Thanks for investigating this. I'm going to merge your contribution again using an alternative route.

**Comment by monsieurp** [comment]
No wait, scratch that. My pram scripts relies on the `git am` command to merge PRs in the form of patches. Apparently, `git am` strips CR automatically and this behaviour is default. There's an option to avoid that: `--keep-cr`.

**Pull request closed without merging**